### PR TITLE
.gitattributes - diff=ruby, eol, rubocop_bundler.yml - eol

### PR DESCRIPTION
**Pull request by MSP-Greg**
## Description:

.gitattributes- text eol=lf, add diff=ruby for several directories

rubocop_bundler.yml - set Layout/EndOfLine to match RubyGems (EOL should be lf)

## What was the end-user or developer problem that led to this PR?

Running RuboCop on Bundler showed errors for **every.file.in.bundler**.

## What is your fix for the problem, implemented in this PR?

see Desc

______________

# Tasks:

- [x] Describe the problem / feature
- [ ] Write tests
- [x] Write code to solve the problem
- [ ] Get code review from coworkers / friends

I will abide by the [code of conduct](https://github.com/rubygems/rubygems/blob/master/CODE_OF_CONDUCT.md).

### CODE REVIEW

File: .gitattributes [hunk 1]
```diff
@@ -0,0 +1,9 @@
+# git log -L:<method>:<file>
+*.rb diff=ruby
+*.gemspec diff=ruby
+bin/* diff=ruby
+bundler/bin/* diff=ruby
+bundler/exe/* diff=ruby
```
Comment (deivid-rodriguez) [inline .gitattributes:6]: All the above seems completely unrelated to this changeset? If it's indeed unrelated, please drop it and explain why you need this on a separate PR.

Or at least split it into a different commit, and explain in the commit message why you need this. I've never seen these configuration on any ruby gem.

File: .gitattributes [hunk 2]
```diff
@@ -0,0 +1,9 @@
...
+
+# Auto detect text files and perform LF normalization
+* text eol=lf
```
Comment (deivid-rodriguez) [inline .gitattributes:9]: I guess you need this for windows, sounds fine :+1:.

File: .rubocop_bundler.yml [hunk 3]
```diff
@@ -297,6 +297,7 @@ Layout/EndAlignment:
 
 Layout/EndOfLine:
   Enabled: true
+  EnforcedStyle: lf
```
Comment (deivid-rodriguez) [inline .rubocop_bundler.yml:300]: :+1:

**Comment by deivid-rodriguez** [comment]
Also, for future PRs, could you give them a title with "human words"? The files being edited, and the changes being made are already part of the commits, so repeating that doesn't add much value. Thanks :pray:.

File: .gitattributes (hunk 1 above)
Comment (MSP-Greg) [reply .gitattributes:6]: https://github.com/ruby/ruby/blob/master/.gitattributes

https://github.com/rails/rails/blob/master/.gitattributes

Comment (deivid-rodriguez) [reply .gitattributes:6]: That's nice, I didn't know about it. I'm putting 

```
*.rb diff=ruby
*.gemspec diff=ruby
```

in my global `~/.git/config/attributes` so I can get the candy without having to add PRs to every project :)

Anyways, I'm good with this :+1:. Will you split this to a separate commit or PR?

Comment (MSP-Greg) [reply .gitattributes:6]: > Will you split this to a separate commit or PR?

Given that this has no affect on the anything but local use (git & RuboCop), I really don't see a need to run CI on it again.  If you really want, I can split it into two commits...  It's six lines excluding comments...

Comment (deivid-rodriguez) [reply .gitattributes:6]: It seems contradictory to me that you want to add changes so that we get a better git history, but refuse to split your commits so that we get a better git history. Either you care about git history, or you don't. So please, either remove the 6 lines, or split them to a separate commit.

Comment (MSP-Greg) [reply .gitattributes:6]: Done - two commits

Comment (deivid-rodriguez) [reply .gitattributes:6]: @MSP-Greg Thanks for splitting the commits. Part of the motivation of splitting unrelated changes into separate commits is to explain the reasons for the changes in the commit messages, so that a future reader can understand the _why_. The commit itself already contains the file being changed, and the changes being made, so please in the future add meaningful commit messages. Thanks!
